doc/dev/error-message.md

```diff
@@ -0,0 +1,20 @@
+# Error message
+
+Elektra has a lot of developers and hence the error messages vary in how they are written.
+This guideline should unify the message format so that users will get consistent error messages.
+
+- All error messages start with a capital letter. The only exception are quoted messages at the start such as "'xxxx-xxxx' is an invalid syntax".
+  If you reference a method name such as `kdbSet()` in the first word, surround it with single quotes.
+  Try to avoid variable strings at the start of the sentence ("%s"-strings) if possible because you cannot guarantee capital casing.
+- **Always** provide the errno if possible. If possible use `strerror_r` for writing out the message because it is thread safe.
+- If you can provide an errno, write `"Reason: %s", strerror(errno)` at the end of the error message. Also end the preceding sentence with a dot.
```

Comment:
Again: strerror should not be recommended to be used.

Reply:
So should we completely avoid errno now as long as no thread safe possibility is there or what exactly should be done?

Reply:
Ideally you would immediately fix it as the strerror are very prominent in your big PR.

Alternatively, you can fix it later (as it should be quite easy to grep for them and remove them all).

For this guideline, I would say something more general anyway, something like:

"Always report all information which could be useful for the user. For some error types, templates are provided to not forget something."

See also #2697

Reply:
I do not want to incorporate another change into the big PR. I might do it later.
I will change the sentence accordingly